src/cript/api/exceptions.py

```diff
@@ -110,10 +110,11 @@ class CRIPTAPISaveError(CRIPTException):
     http_code: str
     api_response: str
 
-    def __init__(self, api_host_domain: str, http_code: str, api_response: str):
+    def __init__(self, api_host_domain: str, http_code: str, api_response: str, pre_saved_nodes: Optional[Set[str]] = None):
```

Comment:
why are we passing in `pre_saved_nodes`?

how come we are passing it in but it's not within the error message?

Reply:
we add this to the exception, because later we can catch this exception and delete all of the pre-saved nodes.
This ensures, that we don't leave orphaned nodes from a failed save attempt.
That is already implemented as far as possible.
(We just need a delete end point.)

Reply:
okay I get what you're doing. 
I don't know a way to currently upgrade this, but in the future we will try to see how we can improve it because the code is getting tangled up and will be hard to update one part without another part breaking, but lets just get it working for now and we can come back to it and make it better